src/control/server/mgmt_system.go

```diff
@@ -841,3 +841,61 @@ func (svc *mgmtSvc) SystemErase(ctx context.Context, pbReq *mgmtpb.SystemEraseRe
 	svc.eraseAndRestart(true)
 	return pbResp, nil
 }
+
+// SystemCleanup implements the method defined for the Management Service.
+//
+// Signal to the data plane to find all resources associted with a given machine
+// and release them. This includes releasing all container and pool handles associated
+// with the machine.
+//
+func (svc *mgmtSvc) SystemCleanup(ctx context.Context, req *mgmtpb.SystemCleanupReq) (*mgmtpb.SystemCleanupResp, error) {
+	if err := svc.checkReplicaRequest(req); err != nil {
+		return nil, err
+	}
+	svc.log.Debugf("Received SystemCleanup RPC: %+v", req)
+
+	if req.Machine == "" {
+		return nil, errors.New("SystemCleanup requires a machine name.")
+	}
+
+	if len(req.Machine) > 64 {
+		return nil, errors.New("Machine Name must be 64 characters or less.")
+	}
+
+	psList, err := svc.sysdb.PoolServiceList()
+	if err != nil {
+		return nil, err
+	}
+
+	resp := new(mgmtpb.SystemCleanupResp)
+	for _, ps := range psList {
+		// Use our incoming request and just replace the uuid on each iteration
+		evictReq := &mgmtpb.PoolEvictReq{
```

Comment:
so rather than recreating the request each time, create the request outside the loop and reuse by just updating the uuid field on the existing request

Reply:
Fixed. Not sure the impact it will have performance/memory wise but I made the change.